healthcheck/healthcheck.go

```diff
@@ -0,0 +1,101 @@
+package healthcheck
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cloudwan/gohan/db"
+	"github.com/cloudwan/gohan/db/transaction"
+	"github.com/cloudwan/gohan/log"
+	"github.com/cloudwan/gohan/sync"
+	"github.com/cloudwan/gohan/util"
+)
+
+const (
+	healthCheckKey        = "healthcheck"
+	healthCheckEnabledKey = healthCheckKey + "/enabled"
+	healthCheckEtcdKey    = healthCheckKey + "/etcd_key"
+	healthCheckAddressKey = healthCheckKey + "/address"
+	healthCheckTimeoutKey = healthCheckKey + "/timeout"
+	defaultEtcdKey        = "/gohan"
+	defaultTimeoutInMs    = 5000
+)
+
+type HealthCheck struct {
+	DataStore db.DB
+	Etcd      sync.Sync
+	etcdKey   string
+	address   string
+	timeout   time.Duration
+	logger    log.Logger
+}
+
+func NewHealthCheck(db db.DB, etcd sync.Sync, serverAddress string, config *util.Config) *HealthCheck {
+	if !config.GetBool(healthCheckEnabledKey, false) {
+		return nil
+	}
+	healthCheckAddress, err := getHealthCheckAddress(serverAddress, config)
+	if err != nil {
+		panic(err)
+	}
+	timeout := time.Duration(config.GetInt(healthCheckTimeoutKey, defaultTimeoutInMs)) * time.Millisecond
```

Comment:
actually, I won't mind if you borrow `GetDuration` from https://github.com/cloudwan/gohan/pull/715/files#diff-d891f1defb2f0e4c0d8b9494144c191aR167 ;)

Reply:
nice - stolen